### PR TITLE
add scala-styled lambdas rendering

### PR DESCRIPTION
**Pull request by Ivoyaa**

### CODE REVIEW

File: izumi-reflect/izumi-reflect/src/main/scala/izumi/reflect/macrortti/LTTRenderables.scala [hunk 1]
```diff
@@ -208,4 +208,31 @@ object LTTRenderables {
...
+    override implicit lazy val r_LambdaParameter: Renderable[LambdaParameter] = new Renderable[LambdaParameter] {
+      override def render(value: LambdaParameter): String = {
+        s"_"
+      }
+    }
+
+    override implicit lazy val r_TypeParam: Renderable[TypeParam] = new Renderable[TypeParam] {
+      override def render(value: TypeParam): String =
+        value.ref match {
+          case n: NameReference if n.symName.name.forall(_.isDigit) =>
```
Comment (pshirshov) [inline izumi-reflect/izumi-reflect/src/main/scala/izumi/reflect/macrortti/LTTRenderables.scala]: This is a heuristic which may fail. We need to preserve the context here and check the names against it. Also we might want to add more metadata into the model to simplify the logic here but that's undesirable.

**Review by pshirshov — COMMENTED** [review]
There is another thing which is still missing. Currently we format prefixes and names in our own non-scala notation (there are some additional symbols, etc) we should take care of that aspect too.

File: izumi-reflect/izumi-reflect/src/main/scala/izumi/reflect/macrortti/LTTRenderables.scala (hunk 1 above)
Comment (Ivoyaa) [reply izumi-reflect/izumi-reflect/src/main/scala/izumi/reflect/macrortti/LTTRenderables.scala]: Added `LambdaParamNameMaker` to hold the logic for both creating param names and checking if a string is a param name in one place. Doesn't seem to me like a clean solution but can't think of any others except extra metadata in model

**Comment by pshirshov** [comment]
This seems to be working, though it still looks like an ad-hoc solution.

I would try one of two options:

1. We might add metadata into the model, why not. Most likely that should be a flag in `NameReference` or (better) a separate branch `LambdaParam` in `SymName`.
2. Instead of using implicit resolution in `LTTRenderables` we might write a recursive traverser which would preserve the context on the stack. I think this solution should be much easier to implement (and it won't break bincompat).

File: izumi-reflect/izumi-reflect/src/main/scala/izumi/reflect/macrortti/LTTRenderables.scala [hunk 2]
```diff
@@ -227,7 +230,62 @@ object LTTRenderables {
...
+            rr
+        }
+      }
+    }
+
+    override implicit lazy val r_Boundaries: Renderable[Boundaries] = new Renderable[Boundaries] {
+      override def render(value: Boundaries): String = value match {
+        case Boundaries.Defined(bottom, top) =>
+          (bottom.render(), top.render()) match {
+            case (bottomRendered, topRendered) if bottomRendered == "scala.Nothing" => s"<: $topRendered"
```
Comment (pshirshov) [inline izumi-reflect/izumi-reflect/src/main/scala/izumi/reflect/macrortti/LTTRenderables.scala]: This is kinda dirty. We already have all the necessary constants in `LightTypeTagInheritance`, please reuse them.

Comment (Ivoyaa) [reply izumi-reflect/izumi-reflect/src/main/scala/izumi/reflect/macrortti/LTTRenderables.scala]: Yep, removed my hardcode

**Pull request merged**